addons/udes_stock/models/stock_picking_batch.py

```diff
@@ -1017,21 +1017,43 @@ def reserve_pallet(self, pallet_name):
         Raises a ValidationError if the pallet is already reserved for another
         batch.
         """
+        Picking = self.env['stock.picking']
         PickingBatch = self.env['stock.picking.batch']
 
         self.ensure_one()
 
-        conflicting_batch = PickingBatch.search([
-            ('id', '!=', self.id),
-            ('state', '=', 'in_progress'),
-            ('u_last_reserved_pallet_name', '=', pallet_name),
-        ])
+        reserve_pallet_per_picking = self.picking_ids.filtered('picking_type_id.u_reserve_pallet_per_picking')
+
+        if reserve_pallet_per_picking:
+            conflicting_picking = Picking.search([
+                ('u_reserved_pallet', '=', pallet_name),
+                ('state', 'not in', ['draft', 'cancelled', 'done'])
+            ])
+            conflicting_batch = conflicting_picking.batch_id
```

Comment:
We might want to raise the error with a conflicting picking instead of the batch. So probably two different error messages.

Reply:
New exception message added